scripts/helpers/utilities.mk

```diff
@@ -440,16 +443,23 @@ define prepare_loader_cache
         cat "$(DYNAMIC_LOAD)/sys32.ld.conf" >> "$(DYNAMIC_LOAD)/ld.so.conf"; \
     fi; \
     grep -E -v "^$$" "$(DYNAMIC_LOAD)/ld.so.conf" > "$(AT_DEST)/etc/ld.so.conf"; \
-    if [[ $1 -eq 2 ]]; then \
-        rm -rf $(AT_DEST)/syslib-override; \
-    else \
-        for cpu in $(BUILD_ACTIVE_MULTILIBS); do \
-            mkdir -p $(AT_DEST)/syslib-override/"$${cpu}"; \
-            (\cd $(AT_DEST)/lib64 && \
-             find . -maxdepth 1 -type f \
-                  -exec cp -p '{}' $(AT_DEST)/syslib-override/"$${cpu}"/'{}' \; \
-            ); \
-        done; \
+    if [[ "$(AT_MAJOR_VERSION)" > 15 ]]; then \
+        if [[ $1 -eq 2 ]]; then \
+            find "$(AT_DEST)/lib64/glibc-hwcaps" -type f \
+                 ! -newer "$(AT_DEST)/lib64/syslib-override" \
+                 -exec rm -f '{}' \;; \
+            rm -f $(AT_DEST)/lib64/syslib-override; \
+        else \
+            for cpu in $(BUILD_ACTIVE_MULTILIBS); do \
+                mkdir -p $(AT_DEST)/lib64/glibc-hwcaps/"$${cpu}"; \
+                (\cd $(AT_DEST)/lib64 && \
+                 find . -maxdepth 1 -type f \
+                     -exec cp --preserve=mode,ownership '{}' \
+                              $(AT_DEST)/lib64/glibc-hwcaps/"$${cpu}"/'{}' \; \
+                ); \
+            done; \
+            touch "$(AT_DEST)/lib64/syslib-override"; \
```

Comment:
This is turning syslib-override into a file. What does this file do?
Is it being used just to indicate which files should be removed?

Reply:
> This is turning syslib-override into a file. What does this file do? Is it being used just to indicate which files should be removed?

Correct. Anything with a timestamp _before_ the timestamp of this file is a fake, and needs to be removed. Not every file has an optimized version, so keeping track explicitly is error-prone. This is a fairly simple shortcut.